FreeRTOS_RA.c

```diff
@@ -71,6 +71,37 @@
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Find a link-local address that is bound to a given interface.
+ *
+ * @param[in] pxInterface: The interface for which a link-local address is looked up.
+ * @param[out] pxIPAddress: The IP address will be copied to this parameter.
+ *
+ * @return pdPASS in case a link-local address was found, otherwise pdFAIL.
+ */
+    static BaseType_t xGetLinkLocalAddress( NetworkInterface_t * pxInterface,
+                                            IPv6_Address_t * pxAddress )
+    {
+        BaseType_t xResult = pdFAIL;
+        NetworkEndPoint_t * pxEndPoint;
+
+        for( pxEndPoint = FreeRTOS_FirstEndPoint( pxInterface );
+             pxEndPoint != NULL;
+             pxEndPoint = FreeRTOS_NextEndPoint( pxInterface, pxEndPoint ) )
+        {
+            if( ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 0 ] == 0xfeU ) &&
+                ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 1 ] == 0x80U ) )
```

Comment:
Do these warrant a comment? One might wonder what `0xfe` and `0x80` is about.

```suggestion
            /* fe80:: is the beginning of a link local address. Check if the IP address of this endpoint
             * matches with the link local IP address. */
            if( ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 0 ] == 0xfeU ) &&
                ( pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 1 ] == 0x80U ) )
```

Reply:
Yes please all magic numbers should have a comment